src/main/java/org/trigon/hardware/rev/spark/SparkMotor.java

```diff
@@ -1,134 +1,184 @@
     }
 
+    /**
+     * Processes the inputs of the motor.
+     */
     public void update() {
         Logger.processInputs("Motors/" + motorName, motorInputs);
+        motorIO.updateSimulation();
     }
 
     public int getID() {
         return id;
     }
 
-    public void registerSignal(SparkSignal signal) {
-        this.registerSignal(signal, false);
+    /**
+     * Registers a threaded signal to be logged from the motor.
+     * Threaded signals use threading to process certain signals separately. Should be used for signals that need to be updated at a high frequency such as odometry.
+     *
+     * @param signal the signal to be registered
+     */
+    public void registerThreadedSignal(SparkSignal signal) {
+        final SparkStatusSignal statusSignal = signal.getStatusSignal(motorIO.getMotor(), motorIO.getEncoder());
+        motorInputs.registerThreadedSignal(statusSignal);
     }
 
-    public void registerSignal(SparkSignal signal, boolean isThreaded) {
+    /**
+     * Registers a signal to be read from the motor.
+     *
+     * @param signal the signal to be read
+     */
+    public void registerSignal(SparkSignal signal) {
         final SparkStatusSignal statusSignal = signal.getStatusSignal(motorIO.getMotor(), motorIO.getEncoder());
-        if (isThreaded)
-            motorInputs.registerThreadedSignal(statusSignal);
-        else
-            motorInputs.registerSignal(statusSignal);
+        motorInputs.registerSignal(statusSignal);
     }
 
+    /**
+     * Gets a signal from the motor.
+     *
+     * @param signal the signal to get
+     * @return the signal
+     */
     public double getSignal(SparkSignal signal) {
         return motorInputs.getSignal(signal.name);
     }
 
-    public void getThreadedSignal(SparkSignal signal) {
-        motorInputs.getThreadedSignal(signal.name);
-    }
-
-    public void setReference(double value, CANSparkBase.ControlType ctrl) {
-        motorIO.setReference(value, ctrl);
-    }
-
-    public void setReference(double value, CANSparkBase.ControlType ctrl, int pidSlot) {
-        motorIO.setReference(value, ctrl, pidSlot);
-    }
-
-    public void setReference(double value, CANSparkBase.ControlType ctrl, int pidSlot, double arbFeedForward) {
-        motorIO.setReference(value, ctrl, pidSlot, arbFeedForward);
-    }
-
-    public void setReference(double value, CANSparkBase.ControlType ctrl, int pidSlot, double arbFeedForward, SparkPIDController.ArbFFUnits arbFFUnits) {
-        motorIO.setReference(value, ctrl, pidSlot, arbFeedForward, arbFFUnits);
-    }
-
-    public void setPeriodicFramePeriod(CANSparkLowLevel.PeriodicFrame frame, int periodMs) {
-        motorIO.setPeriodicFramePeriod(frame, periodMs);
-    }
-
+    /**
+     * Gets a threaded signal from the motor.
+     * Threaded signals use threading to process certain signals separately. Should be used for signals that need to be updated at a high frequency such as odometry.
```

Comment:
Threaded signals use threading to process certain signals separately at a faster rate.
Keep the jdoc consistent across all threaded stuff